lib/srv/desktop/rdp/rdpclient/src/util.rs

```diff
@@ -20,9 +24,48 @@
 /// UTF-16LE encoded Vec<u8>, which is useful in cases where we want
 /// to handle some data in the code as a &str (or String), and later
 /// convert it to RDP's preferred format and send it over the wire.
-pub fn to_unicode(s: &str) -> Vec<u8> {
-    let mut buf: Vec<u8> = s.encode_utf16().flat_map(|v| v.to_le_bytes()).collect();
-    let mut null_terminator: Vec<u8> = vec![0, 0];
-    buf.append(&mut null_terminator);
+pub fn to_unicode(s: &str, with_null_term: bool) -> Vec<u8> {
+    let mut buf = WString::<LE>::from(s).as_bytes().to_vec();
```

Comment:
Was there something broken about the previous approach with `flat_map`?

Reply:
No. The reason I changed this is because I decided that it was best to add that `utf16string` library for `from_unicode` and so I used it here too for consistency's sake.